rover/src/owr_drive_controls/src/crabSteer.cpp

```diff
@@ -0,0 +1,119 @@
+/*
+ * Implementation of crab steering logic
+ */
+
+#include "crabSteer.hpp"
+#include <ros/ros.h>
+
+#define ROVER_CENTRE_2_WHEEL_Y 0.40023
+#define BACK_WHEEL_SPAN 0.80273
+#define HALF_ROVER_WIDTH_X .27130
+#define FRONT_W_2_BACK_W_X 0.54216
+#define VEL_ERROR 0.01
+
+#define DEG90 M_PI_2
+
+#define RIGHT 1
+#define LEFT -RIGHT
+
+crabMotorVels steerCrab(crabMotorVels,
+                     double,
+                     double,
+                     double,
+                     double,
+                     double,
+                     int dir);
+
+int getDir(double);
+
+/**
+ * sqrt(x^2 + y^2)
+ */
+int getHypotenuse(int x, int y) {
+    return sqrt(pow(x, 2) + pow(y, 2));
+}
+
+
+int getVelMagnitude(const geometry_msgs::Twist * velMsg) {
+    return fabs(getHypotenuse(velMsg->linear.x, velMsg->linear.y));
+}
+
+crabMotorVels steerCrab(crabMotorVels vels,
+                     double closeFrontV,
+                     double farFrontV,
+                     double farBackV,
+                     double closeBackV,
+                     double turnAngle,
+                     int dir) {
+
+    crabMotorVels output = vels;
```

Comment:
I don't think you need this line/parameter?

Reply:
It is not clear which line/parameter you are referring to. Are you talking about `vels`?